File: c3-prompt-script/prompt_client.py
import requests
import json
import logging
import os
from typing import Dict, Optional, Any

logger = logging.getLogger(__name__)

class PromptClient:
    """Client for sending prompts to Comput3 API"""
    
    def __init__(self, api_key: str):
        """Initialize with Comput3 API key"""
        self.api_key = api_key
        self.api_url = "https://app.comput3.ai/0/api/generate"
        
        if not api_key:
            logger.error("🔑 C3 API key is not provided. Please set your C3_API_KEY in .env file.")
    
    def get_headers(self) -> Dict[str, str]:
        """Get the headers for API requests"""
        return {
            "Authorization": f"Bearer {self.api_key}",
            "Content-Type": "application/json"
        }
    
    def send_prompt(self, prompt: str, model: str = "llama3:70b") -> Optional[Dict[str, Any]]:
        """Send a prompt to the C3 API and return the response"""
        if not self.api_key:
            logger.error("❌ Cannot send prompt: C3 API key is missing")
            return None
        
        try:
            payload = {
                "model": model,
                "prompt": prompt,
                "stream": False
            }
            
            logger.info(f"🚀 Sending prompt to model: {model}")
            response = requests.post(
                self.api_url,
                headers=self.get_headers(),
                json=payload
            )
            
            if response.status_code != 200:
                logger.error(f"❌ Failed to send prompt: {response.status_code} - {response.text}")
                return None
            
            logger.info("✅ Successfully received response from model")
            return response.json()
            
        except Exception as e:
            logger.error(f"❌ Error sending prompt: {str(e)}")
            return None
    
    def save_response_to_file(self, response: Dict[str, Any], prompt_num: int, output_dir: str = "./results") -> Optional[str]:
        """Save the model response to a file"""
        # Create output directory if it doesn't exist
        os.makedirs(output_dir, exist_ok=True)
        
        try:
            # Create a filename based on prompt number
            output_file = os.path.join(output_dir, f"response_{prompt_num}.txt")
            
            # Extract the response text from the API response
            response_text = response.get("response", "")
            
            # Write to file
            with open(output_file, "w", encoding="utf-8") as f:
                f.write(response_text)
                
            return output_file
            
        except Exception as e:
            logger.error(f"❌ Error saving response to file: {str(e)}")
            return None

File: 003-prompt-script-nodejs/promptClient.js
const axios = require('axios');
const fs = require('fs');
const path = require('path');

class PromptClient {
  /**
   * Initialize with Comput3 API key
   * @param {string} apiKey - The Comput3 API key
   */
  constructor(apiKey) {
    this.apiKey = apiKey;
    this.apiUrl = 'https://app.comput3.ai/0/api/generate';
    
    if (!apiKey) {
      console.error('🔑 C3 API key is not provided. Please set your C3_API_KEY in .env file.');
    }
  }

  /**
   * Get the headers for API requests
   * @returns {Object} - The headers for API requests
   */
  getHeaders() {
    return {
      'Authorization': `Bearer ${this.apiKey}`,
      'Content-Type': 'application/json'
    };
  }

  /**
   * Send a prompt to the C3 API and return the response
   * @param {string} prompt - The prompt to send
   * @param {string} model - The model to use (default: "llama3:70b")
   * @returns {Promise<Object|null>} - The API response or null if there was an error
   */
  async sendPrompt(prompt, model = 'llama3:70b') {
    if (!this.apiKey) {
      console.error('❌ Cannot send prompt: C3 API key is missing');
      return null;
    }
    
    try {
      const payload = {
        model: model,
        prompt: prompt,
        stream: false
      };
      
      console.log(`🚀 Sending prompt to model: ${model}`);
      
      const response = await axios.post(
        this.apiUrl,
        payload,
        { headers: this.getHeaders() }
      );
      
      if (response.status !== 200) {
        console.error(`❌ Failed to send prompt: ${response.status} - ${response.statusText}`);
        return null;
      }
      
      console.log('✅ Successfully received response from model');
      return response.data;
    } catch (error) {
      console.error(`❌ Error sending prompt: ${error.message}`);
      return null;
    }
  }

  /**
   * Save the model response to a file
   * @param {Object} response - The API response
   * @param {number} promptNum - The prompt number
   * @param {string} outputDir - The directory to save the response to (default: "./results")
   * @returns {string|null} - The path to the saved file or null if there was an error
   */
  saveResponseToFile(response, promptNum, outputDir = './results') {
    // Create output directory if it doesn't exist
    if (!fs.existsSync(outputDir)) {
      fs.mkdirSync(outputDir, { recursive: true });
    }
    
    try {
      // Create a filename based on prompt number
      const outputFile = path.join(outputDir, `response_${promptNum}.txt`);
      
      // Extract the response text from the API response
      const responseText = response.response || '';
      
      // Write to file
      fs.writeFileSync(outputFile, responseText, 'utf8');
      
      return outputFile;
    } catch (error) {
      console.error(`❌ Error saving response to file: ${error.message}`);
      return null;
    }
  }
}

module.exports = PromptClient; 